lib/karafka/routing/subscription_groups_builder.rb

```diff
@@ -39,6 +39,13 @@ def call(topics)
           .map { |value| value.map(&:last) }
           .map { |topics_array| Routing::Topics.new(topics_array) }
           .map { |grouped_topics| SubscriptionGroup.new(@position += 1, grouped_topics) }
+          .tap do |subscription_groups|
+            subscription_groups.each do |subscription_group|
+              subscription_group.topics.each do |topic|
+                topic.subscription_group = subscription_group
```

Comment:
Shouldn't this be `topic.subscription_group_id = subscription_group_id`?

Reply:
No. Here we assign the final materialized subscription group. Id is an intermediate "notion" of groups when routing are being built.